cmd/operator-sdk/generate/csv.go

```diff
@@ -27,17 +27,19 @@ import (
 )
 
 type csvCmd struct {
-	csvVersion     string
-	csvChannel     string
-	fromVersion    string
-	operatorName   string
-	outputDir      string
-	deployDir      string
-	apisDir        string
-	crdDir         string
-	updateCRDs     bool
-	defaultChannel bool
-	makeManifests  bool
+	csvVersion       string
+	csvChannel       string
+	fromVersion      string
+	operatorName     string
+	outputDir        string
+	deployDir        string
+	apisDir          string
+	crdDir           string
+	updateCRDs       bool
+	defaultChannel   bool
+	makeManifests    bool
+	interactive      bool
+	promptpreference bool
```

Comment:
This works fine for now. Since the interactive library is reusable, it would be nice to also implement a "level of interactive prompt" enum in the future:

```go
type InteractiveLevel int

const (
	// User has not turned interactive mode on or off, default to off.
	InteractiveSoftOff InteractiveLevel = iota
	// User has explicitly turned interactive mode off.
	InteractiveHardOff
	// User only wants required interactive prompts.
	InteractiveOnRequired
	// User wants normal interactive prompts.
	InteractiveOnAll
)
```

This enum would simplify fields in `BundleGenerator` for example. Just a thought for a follow-up.

Reply:
Agreed. Further when it is InteractiveSoftOff we can look for baseCSV and provide prompts accordingly. I assume `InteractiveOnAll` is for future use, where we have optional prompts configured.

Reply:
We have both optional and required prompts now right?

Reply:
Yes, but that appears anyway when the user switches the interactive option. Its just that user need not provide any value. In that case, we would not need `InteractiveOnAll` option.